developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,129 @@
+.. _uuid:
+
+UUID Usage in SOF
+#################
+
+Why UUID Needed
+***************
+
+To develop a new audio signal processing component, we traditionally
+implement the component driver with a new unique component type
+introduced. The pain points with this method include how to keep the
+ABI (Application Binary Interface) aligned for the topology file, the
+driver, and the firmware. And if unaligned, how should we make them
+backward compatible. For example, When adding a new component, we could
+have 8 types of combination of topology/driver/firmware versions with
+either the new component is supported or not in each
+topology/driver/firmware part. Even more, if enumerated type is used
+for the component type and the sequence in the enumerate list
+(the value) is changed during an update, there could be component type
+collision if versions are not aligned.
+
+UUIDs (Universally Unique Identifiers) provide a more scalable and
+collision-free way of component identification. UUIDs are used as the
+standard interface by all users of the SOF firmware, including the
+tracing subsystem, the topology .m4 files, and the Linux topology
+driver.
+
+Allocating a new UUID for the new added component is recommended, as
+the component type will be replaced by UUID in the IPC structures,
+in the future. For the whole SOF stack, the usage of the UUID should
+follow rules as below:
```

Comment:
again you are using recommended and 'should' when it's now mandatory. You have to be consistent with the language used between paragraphs.